Account

HamidNiakan opened the pull request.

HamidNiakan committed:
config route api,creating http Controller as AccountController,creatig Resource for return response json and also creating FormRequest for inserting,updating and deleting account

HamidNiakan committed:
checking authentication user when creating new account , update ,delete and filter

HamidNiakan committed:
managment transaction with feature test

yeganemehr commented on src/Http/Controllers/AccountController.php:
```diff
@@ -0,0 +1,92 @@
+class AccountController extends Controller {
+	public AccountManager $account_manager;
+	
+	public function __construct ( AccountManager $account_manager ) {
```
```suggestion
	public function __construct ( protected AccountManager $accountManager ) {
```

We use php 8.1, you can code less.
for naming variables use camelCase

yeganemehr commented on src/Http/Controllers/AccountController.php:
```diff
@@ -0,0 +1,92 @@
+	}
+	
+	public function create ( CreateNewAccountRequest $request ) {
+		if ( !auth()->check() ) {
```
Do it as "auth" middleware in routes

yeganemehr commented on src/Http/Controllers/AccountController.php:
```diff
@@ -0,0 +1,92 @@
+		$meta = $request->get('meta');
+		$user_id = auth()->user() ? auth()->user()->id : null;
+		$account = $this->account_manager->create($title , $currency_id , $user_id , AccountStatus::ACTIVE , $can_send , $can_receive , $meta);
+		$account = AccountResource::make($account);
```
just return the resource

yeganemehr commented on src/routes/api.php:
```diff
@@ -0,0 +1,21 @@
+
+use Illuminate\Support\Facades\Route;
+
+/*
```
Remove educational comments.

yeganemehr commented on src/routes/api.php:
```diff
@@ -0,0 +1,21 @@
+*/
+Route::controller(\dnj\Account\Http\Controllers\AccountController::class)
+	 ->group(function () {
+		 Route::post('create' , 'create');
```
Read about RESTFUL url template
search => GET /accounts
create => POST /accounts
update => PUT /accounts/{accountId}
delete => DELETE /accounts/{accountId}

HamidNiakan committed:
creating api docs ,refactoring AccountServiceProvider,remving extra code in AccountController,adding middleware auth for api route

HamidNiakan committed:
rebuilding api docs for account,create model User and UserFactory,rebuild Feature test for account and also refactoring code and removing extra code

yeganemehr committed:
Refactoring

yeganemehr committed:
Fix codestyle

yeganemehr committed:
Fix helper methods for tests

yeganemehr committed:
add github workflow for running tests and check code coverage

yeganemehr commented on src/Http/Controllers/AccountController.php:
```diff
@@ -0,0 +1,93 @@
+    {
+    }
+
+    public function index(AccountSearchRequest $request)
```
please test this function

HamidNiakan replied:
done

yeganemehr commented on src/Http/Controllers/TransactionController.php:
```diff
@@ -0,0 +1,73 @@
+    {
+    }
+
+    public function index(Account $account, TransactionSearchRequest $request)
```
please test this method.

HamidNiakan replied:
done

yeganemehr committed:
move routes out of src directory

HamidNiakan committed:
Implemented the filter for the account and also wrote the test for it

HamidNiakan committed:
changing apiDoc

HamidNiakan committed:
completing account Filter feature test according to balance and created_at
completing transaction filter feature test according to amount and created_at
fix up response  account and transaction filter
fix up route account and transaction
